ERCS/erc-7683.md

```diff
@@ -24,153 +24,342 @@ By implementing a standard, cross-chain intents systems can interoperate and sha
+/// @notice An implementation-generic representation of an order intended for filler consumption
 /// @dev Defines all requirements for filling an order by unbundling the implementation-specific orderData.
 /// @dev Intended to improve integration generalization by allowing fillers to compute the exact input and output information of any order
 struct ResolvedCrossChainOrder {
-	/// @dev The contract address that the order is meant to be settled by.
-	address settlementContract;
-	/// @dev The address of the user who is initiating the swap
-	address swapper;
-	/// @dev Nonce to be used as replay protection for the order
-	uint256 nonce;
+	/// @dev The address of the user who is initiating the transfer
+	address user;
 	/// @dev The chainId of the origin chain
-	uint32 originChainId;
-	/// @dev The timestamp by which the order must be initiated
-	uint32 initiateDeadline;
+	uint64 originChainId;
+	/// @dev The timestamp by which the order must be opened
+	uint32 openDeadline;
 	/// @dev The timestamp by which the order must be filled on the destination chain(s)
 	uint32 fillDeadline;
 
-	/// @dev The inputs to be taken from the swapper as part of order initiation
-	Input[] swapperInputs;
-	/// @dev The outputs to be given to the swapper as part of order fulfillment
-	Output[] swapperOutputs;
-	/// @dev The outputs to be given to the filler as part of order settlement
-	Output[] fillerOutputs;
-}
-
-/// @notice Tokens sent by the swapper as inputs to the order
-struct Input {
-	/// @dev The address of the ERC20 token on the origin chain
-	address token;
-	/// @dev The amount of the token to be sent
-	uint256 amount;
+	/// @dev The max outputs that the filler will send. It's possible the actual amount depends on the state of the destination
+	///      chain (destination dutch auction, for instance), so these outputs should be considered a cap on filler liabilities.
+	Output[] maxSpent;
+	/// @dev The minimum outputs that must to be given to the filler as part of order settlement. Similar to maxSpent, it's possible
+	///      that special order types may not be able to guarantee the exact amount at open time, so this should be considered
+	///      a floor on filler receipts.
+	Output[] minReceived;
+	/// @dev Each instruction in this array is parameterizes a single leg of the fill. This provides the filler with the information
+	///      necessary to perform the fill on the destination(s).
+	FillInstruction[] fillInstructions;
 }
 
 /// @notice Tokens that must be receive for a valid order fulfillment
 struct Output {
 	/// @dev The address of the ERC20 token on the destination chain
 	/// @dev address(0) used as a sentinel for the native token
-	address token;
+	bytes32 token;
 	/// @dev The amount of the token to be sent
 	uint256 amount;
 	/// @dev The address to receive the output tokens
-	address recipient;
+	bytes32 recipient;
 	/// @dev The destination chain for this output
-	uint32 chainId;
+	uint64 chainId;
```

Comment:
Note we may want to change this to uint256 to support possible patterns like [ERC7785,](https://github.com/ethereum/ERCs/pull/669/files) which would use a bytes32 identifier instead of integer chain ID's

Reply:
Done